lib/active_admin/filters/active_filter.rb

```diff
@@ -90,16 +90,27 @@ def find_class
         if condition_attribute.klass != resource_class && condition_attribute.klass.primary_key == name.to_s
           condition_attribute.klass
         else
-          assoc = condition_attribute.klass.reflect_on_all_associations.
-            reject { |r| r.options[:polymorphic] }. #skip polymorphic
-            detect { |r| r.foreign_key.to_s == name.to_s }
-          assoc.class_name.constantize if assoc
+          predicate_association.class_name.constantize if predicate_association
```

Comment:
Should this become an `elsif` for easier reading?

Reply:
It was like that previously, so I didn't know if it was convention to avoid if - elsif statements with no else.